include/fplus/side_effects.hpp

```diff
@@ -283,6 +283,46 @@ auto execute_n_times(std::size_t n, const Effect& eff)
     }
 }
 
+// API search type: parallel_for : (Io a, [a]) -> Io ()
+// Runs the function `f` in parallel on all the `idxs_to_iterate` container elements.
+// The function will perform its side effects, and nothing is returned.
+// -
+// Example:
+// -
+// constexpr int rows = 3000, cols = 2000;
+// MandelbrotMat<rows, cols> mandelbrot_mat;
+// auto fill_mandelbrot_row = [& mandelbrot_mat ](int row_idx) {
+//        ...
+// };
+// mandelbrot_mat = MandelbrotMat<rows, cols>();
+// auto rows_idxs = fplus::numbers(0, rows);
+// fplus::parallel_for(fill_mandelbrot_row, rows_idxs);
+template<typename F, typename ContainerIdxs>
+void parallel_for(F f, const ContainerIdxs& idxs_to_iterate)
```

Comment:
I wouldn't even call it `ContainerIdxs& idxs_to_iterate`, but just `Container& xs`. Of course, the elements can be indexes, but they don't have to. :)